docs/source/format/Columnar.rst

```diff
@@ -693,6 +690,16 @@ Only the slot in the array corresponding to the type index is considered. All
 "unselected" values are ignored and could be any semantically correct array
 value.
 
+.. note:: Critically, the sparse union allows shared columns to be reused between union members
+   in the ubiquitous union-of-structs with non-overlapping-fields use case.  For example the union::
+
+       SparseUnion<m1: Struct<i: Int32>,
+                   m2: Struct<i: Int32, f: Float32, s: VarBinary>,
+                   m3: Struct<f: Float32, s: VarBinary>>
+
+   could be backed by just three columns (one for each type) since no union member requires more
+   than one of each.
```

Comment:
I've added a test which passes locally. I don't think it'd be illegal for an IPC writer to reuse a buffer though it's an optimization I wouldn't expect to be useful with any frequency. Should I just remove this note?

Reply:
Hmm, right. But it wouldn't necessarily be reused on the read side, unless the reader actively tries to detect this case?

Reply:
I think this note is confusing as well, because it's suggesting that something that requires a good deal of care can be done without really giving any details about how one might go about doing it. Personally, I think we should probably remove the note entirely.

Reply:
I'll remove it. FTR this note has been there since ARROW-3 and the first writing of the sparse/dense union format spec. @wesm any comment?